services/director/src/simcore_service_director/producer.py

```diff
@@ -44,6 +44,7 @@ async def _check_node_uuid_available(
     log.debug("Checked if UUID %s is already in use", node_uuid)
     # check if service with same uuid already exists
     try:
+        # not filtering by "swarm_stack_name" label because it's safer
         list_of_running_services_w_uuid = await client.services.list(
             filters={"label": "uuid=" + node_uuid}
```

Comment:
the chance is very low but maybe the name of the service should be somehow encoded with the swarm name to prevent it from happening.
In case of staging/production the service it could be possible that staging and production both start a service with same uuid.
and now swarm_stack_name has a non-empty default isn't it?

Reply:
You are correct. And by not adding the label to it, we make sure there is no collision between staging and production.

Reply:
hmm not sure about that. isn't it firing an exception if the uuid is already taken? then the service does not start.

Reply:
You are correct. Thought it would retry automatically.  Will issue a fix

Reply:
I gave it a new thought. It is correct like this.
When creating a new service, in the case where production and staging are mixed together, we have to look out for duplicate uuids and not filter by `swarm_stack_name`.
This will not change what it did before (the exception will occur). It is correct to not filter by the `swarm_stack_name ` label.